### PR TITLE
Create steelseries-rival-110

### PR DESCRIPTION
**Pull request by mucahiddogan**

### CODE REVIEW

**Comment by FFY00** [comment]
Can you confirm everything works as expeted? Are there any features you did not test?

**Comment by mucahiddogan** [comment]
wait just couple of days. I will test again and feedback to you

**Comment by FFY00** [comment]
Do you have any update on this? I just need you to make sure that the devices does work correctly with all features implemented by the driver so that we do not ship broken software to users.

**Comment by mucahiddogan** [comment]
bro. I am not using these now. If I try again. I sware I will add the configuration and will test it:=

**Comment by amfern** [comment]
When you have the time, can you test the LED, DPI and button macros are working as expected for you, we will be able to merge it then

**Comment by iDre4M** [comment]
Mhm, how the things are going with this?

**Comment by perigoso** [comment]
someone with a rival 110 needs to test this, do you have one @iDre4M?

**Comment by iDre4M** [comment]
> someone with a rival 110 needs to test this, do you have one @iDre4M?

Yeah, I do have it at home, but currently I'm on the way so not with me.

**Comment by perigoso** [comment]
Well, then when you have it with you, you can test this, and if everything looks good we can merge it

**Comment by amfern** [comment]
@iDre4M Hi, just checking if you are back from travel and able to test this patch?

**Comment by iDre4M** [comment]
> @iDre4M Hi, just checking if you are back from travel and able to test this patch?

Yeah, I'm back, sorry that haven't replied earlier. Give short instructions how to test it out and I'll let you know, if needed can contact via Discord.

**Comment by RavingSombra** [comment]
Heya,
I ended up finding this right when I was thinking about using libratbag with a custom profile for the Rival 110.
This is the mouse I mainly use. Don't have Linux on my main rig, but I have no worries on using my laptop to validate these changes.

Is there anything specific I need to do while testing?

**Review by RavingSombra — CHANGES_REQUESTED** [review]
Minimum DPI that can be set in SS Engine (on Windows) is 200. The range should be 200-7200 with 100 steps

rival110SSE

**Comment by RavingSombra** [comment]
I've started testing today.
The mouse is recognized and all the buttons work as expected.

Issues:
The mouse is detected twice;
Changing RGB modes or colors isn't working
DPI changes aren't that much noticible (7200 feels nearly the same as 1600, even though on Windows I can tell the difference right away)
Screenshot from 2021-12-29 15-34-59

I'll do some more testing to figure out which of the detected devices really makes changes saved on the mouse firmware

**Comment by staticssleever668** [comment]
@RavingSombra, hi.
Mouse being recognized twice was fixed, but it's not in a release yet.
Could you please try following device file? DPI changing probably would still not work, but it should fix  lighting color and modes.
```
[Device]
# 110; 106.
DeviceMatch=usb:1038:1729;usb:1038:1816
Driver=steelseries
Name=SteelSeries Rival 110/106

[Driver/steelseries]
# Actually 6.
Buttons=0
DeviceVersion=1
DpiRange=200:7200@100
Leds=1
Quirk=Rival100
```

**Comment by RavingSombra** [comment]
I've tested the changes with the latest build ae53f61 and the device file you sent

Mouse showing up duplicated is fixed.
LED is improved. I'm able to change colors but cannot turn the LEDs off (even though in breathing mode it fades from off to full color successfully)

DPI is not working. Every time I plug the mouse in it detects 800DPI in profile 0 and 1600 in profile 1, despite the values set through Steelseries GG in Windows. Also despite what DPI I set on ratbagctl, the info command shows the value I've set but it isn't saved on the mouse, so when I unplug and plug it back in the detected values are 800 and 1600

**Comment by staticssleever668** [comment]
@RavingSombra, I see, thanks for testing, but it's a bit unclear from your message, does DPI changing work? If it does, I think we can add the device file I sent.
Described problems are limitations of this mouse model. There are no ways to read DPI and lighting mode from the mouse (not that I know of). To disable the LED you have to set it to `ON` and set the color to black (this should probably be worked around in the driver...).

**Comment by RavingSombra** [comment]
The DPI settings are kind of dummy.
I can change the speed, but 200-800DPI feels the same. Also anything from 1600 to 7200 DPI feels the same.
Even though the DPI can be changed, I don't consider it's working well enough to consider the mouse as supported (mainly due to the mentioned limitations, including reading/writing to the mouse)

TLDR:
LEDs work totally fine
DPI settings have limited functionality (setting a DPI value only works until the mouse is unplugged or the PC is turned off/restarted)
Mouse buttons cannot be customized (didn't test a config with the 6 buttons tho)

**Comment by staticssleever668** [comment]
I'll close this since it also needs protocol changes on the driver side.
